src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/SourceManifest.cs

```diff
@@ -0,0 +1,102 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+#nullable enable
+namespace Microsoft.DotNet.Darc.Models.VirtualMonoRepo;
+
+public class SourceManifest
+{
+    public ICollection<RepositoryRecord> Repositories { get; set; }
+    public ICollection<SubmoduleRecord> Submodules { get; set; }
+
+    public SourceManifest()
+    {
+        Repositories = new List<RepositoryRecord>();
+        Submodules = new List<SubmoduleRecord>();
+    }
+
+    public void UpdateVersion(string repository, string uri, string sha, string packageVersion)
+    {
+        var repo = Repositories.FirstOrDefault(r => r.Path == repository);
+        if (repo != null)
+        {
+            repo.CommitSha = sha;
+            repo.RemoteUri = uri;
+            repo.PackageVersion = packageVersion;
+        }
+        else
+        {
+            Repositories.Add(new RepositoryRecord 
+            { 
+                Path = repository,
+                CommitSha = sha,
+                RemoteUri = uri,
+                PackageVersion = packageVersion
+            });
+        }
+    }
+
+    public string ToJson()
+    {
+        var options = new JsonSerializerOptions
+        {
+            AllowTrailingCommas = true,
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            WriteIndented = true,
+        };
+
+        return JsonSerializer.Serialize<SourceManifest>(this, options);
```

Comment:
@MilenaHristova one think I just thought of - we will need to serialize the repositories by path ascending. Same for submodules later

Reply:
ok

Reply:
can I store them in a sorted collection like SortedSet?

Reply:
Yes, that's a great solution